### ISSUE TITLE
Entity Service API and REST API both with populate=* do not format data the same way

### ISSUE THREAD

**Issue by VictorienTardif** [strapi/strapi#13101]
### Required System information

- Node.js version: 16.13.1
- NPM version: 8.3.0
- Strapi version: 4.1.7
- Database: mysql
- Operating system: MacOS

### Describe the bug

Let's say we have `articles` with a relation to `article_categories`.
When querying the REST API with `/api/articles?populate=*`, we obtain a response of this shape:
```
{
  data: [
    {
      id: 1,
      attributes: {
        article_categories: {
          data: [{ id: 1 }],
        },
      },
    },
  ],
}
```
But, when using the Entity Service API with: 
```
const articles = await strapi.entityService.findMany('api::article.article', {
   populate: '*',
});
```
we obtain a response of this shape:
```
[
  {
    id: 1,
    article_categories: [{ id: 1 }],
  },
]
```

### Expected behavior

Shouldn't we expect both APIs to format data the same way?

I have a custom controller that uses the entityService to populate data and on the frontend I need to have a constant formatting, either the controller is custom or not.

**Comment by VictorienTardif** [comment]
I also tried to use the controller `transformResponse` function but it is not working neither.

**Comment by derrickmehaffy** [comment]
No not entirely as the entityService is used by various internal systems not just REST. 

Going down the rabbit hole it looks like we hand things off from the entityService -> Sanitize -> TransformResponse -> TransformEntity:

Example using the default find controller (I'm not gonna dive down the sanitizeOutput, if you want to explore that go head)

 - Fetch data: https://github.com/strapi/strapi/blob/23a28790a9546b969bf3f8a675f61d45cb6316fb/packages/core/strapi/lib/core-api/controller/collection-type.js#L24
 - Sanitize: https://github.com/strapi/strapi/blob/23a28790a9546b969bf3f8a675f61d45cb6316fb/packages/core/strapi/lib/core-api/controller/collection-type.js#L25
 - TransformResponse Call: https://github.com/strapi/strapi/blob/23a28790a9546b969bf3f8a675f61d45cb6316fb/packages/core/strapi/lib/core-api/controller/collection-type.js#L27
 - TransformResponse function: https://github.com/strapi/strapi/blob/23a28790a9546b969bf3f8a675f61d45cb6316fb/packages/core/strapi/lib/core-api/controller/transform.js#L16
 - TransformEntity: https://github.com/strapi/strapi/blob/23a28790a9546b969bf3f8a675f61d45cb6316fb/packages/core/strapi/lib/core-api/controller/transform.js#L42
 - And where/how attributes is set:
 
 https://github.com/strapi/strapi/blob/23a28790a9546b969bf3f8a675f61d45cb6316fb/packages/core/strapi/lib/core-api/controller/transform.js#L57-L92

If you really wanted to you could build custom controllers and instead just use the `super.*` methods just like extending the core ones I believe (I haven't tried it but it should work) which would just wrap the controller logic (or various) with your custom code. Depends on what you are trying to do.

**Comment by VictorienTardif** [comment]
Hello @derrickmehaffy, thank you for your answer.

I already went through those functions but I guess I was missing something in the code, I thought the `transformResponse` function was generic but I had to use the controller  `strapi.controller('api::article.article')` to `sanitizeOutput` and `transformResponse` the right way. Not the controller I was in. (I'm adding articles to the component of a single type, so I was in the controller of that single type).

Now I have something working doing:

```
const uid = 'api::article.article';
const controller = strapi.controller(uid);
const results = await strapi.entityService.findMany(uid, {
   populate: '*',
});
const sanitizedResults = await controller.sanitizeOutput(results, ctx);
const transformedResponse = controller.transformResponse(sanitizedResults);
```
`transformedResponse` is now formatted the right way :) 
Thanks!
